src/Core/src/Platform/iOS/AcceleratorExtensions.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+using Foundation;
+using ObjCRuntime;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	internal static class AcceleratorExtensions
+	{
+		internal static UIMenuElement CreateMenuItem(this IMenuFlyoutItem virtualView, IMauiContext mauiContext)
+		{
+			int index = MenuFlyoutItemHandler.menus.Count;
+			UIImage? uiImage = virtualView.Source.GetPlatformMenuImage(mauiContext);
+			var selector = new Selector($"MenuItem{index}:");
+
+			bool selectorFound =
+				MauiUIApplicationDelegate.Current.RespondsToSelector(selector);
+
+			if (!selectorFound)
+			{
+				throw new InvalidOperationException(
+					$"By default we only support 50 MenuItems. You can add more by adding the following code to {MauiUIApplicationDelegate.Current.GetType()}\n\n" +
+					$"[Export(\"MenuItem{index}: \")]\n" +
+					$"internal void MenuItem{index}(UICommand uICommand)\n" +
+					"{\n" +
+					"	uICommand.SendClicked();\n" +
+					"}");
+			}
+
+			var accelerator = virtualView.Accelerators?[0];
+			if (accelerator is null)
+				return virtualView.CreateMenuItemCommand(index, uiImage, selector);
+
+			var key = accelerator.Key;
+			var modifiers = accelerator.Modifiers;
+
+			UIKeyModifierFlags modifierFlags = 0;
+
+			// Single key accelerators (no modifier, i.e. "A") and multi-key accelerators are supported (>=1 modifier and 1 key only, i.e. Ctrl+F, Ctrl+Shift+F)
+			if (modifiers is not null)
+			{
+				var modifiersCount = modifiers.Count();
+
+				if (modifiersCount > 0)
+				{
+					for (int i = 0; i < modifiersCount; i++)
+					{
+						var modifierMask = modifiers.ElementAt(i).ToLowerInvariant();
+
+						switch (modifierMask)
+						{
+							case "ctrl":
+								modifierFlags |= UIKeyModifierFlags.Control;
+								break;
+							case "cmd":
+								modifierFlags |= UIKeyModifierFlags.Command;
+								break;
+							case "alt":
+								modifierFlags |= UIKeyModifierFlags.Alternate;
+								break;
+							case "shift":
+								modifierFlags |= UIKeyModifierFlags.Shift;
+								break;
+						}
+					}
+				}
+			}
```

Comment:
Why are you us ing a for loop with an index in a an Innumerable when you only use the index to access the modifier? Can you simply use a foreach. In that case, you don't even need to access Count, right? Just simply use the Enumerator and loop. Am I missing something that forces you to use the i index?

Reply:
Agree, changed.